File: v6/sprefa-extract/tests/fixtures/df_loops/sample.ts
class Cell {
  value: number;
  constructor(value: number) {
    this.value = value;
  }
}

function allocating(rows: number[], cols: number[]): Cell[] {
  const out: Cell[] = [];
  for (const row of rows) {
    for (const col of cols) {
      const cell = new Cell(add(row, col));
      out.push(cell);
    }
  }
  return out;
}

function plain(limit: number): number {
  let total = 0;
  while (total < limit) {
    total = add(total, 1);
  }
  return add(total, 0);
}

function add(left: number, right: number): number {
  return left + right;
}

File: v6/sprefa-extract/tests/fixtures/df_loops/sample.go
package sample

type Cell struct {
	Value int
}

func Allocating(rows []int, cols []int) []Cell {
	out := []Cell{}
	for _, row := range rows {
		for _, col := range cols {
			cell := Cell{Value: Add(row, col)}
			out = append(out, cell)
		}
	}
	return out
}

func Plain(limit int) int {
	total := 0
	for total < limit {
		total = Add(total, 1)
	}
	return Add(total, 0)
}

func Add(left int, right int) int {
	return left + right
}
